spec/rspec/support/object_inspector_spec.rb

```diff
@@ -92,6 +92,44 @@ def with_date_loaded
             matcher_without_a_description.inspect)
         end
       end
+
+      context "with double objects" do
+        it "displays double name" do
+          double = double('Foo')
+          allow(double).to receive(:name).and_return("Foo")
+          ObjectInspector.register double.class do |double|
+            "Double (#{double.name})"
+          end
+          expect(ObjectInspector.inspect(double)).to eq("Double (Foo)")
+        end
+      end
+
+      context "with arbitary type" do
+        class User
+          attr_accessor :name
+
+          def initialize(name)
+            @name = name
+          end
+        end
+
+        it "works with custom class" do
+          ObjectInspector.register User do |user|
+            "User (#{user.name})"
+          end
+
+          user = User.new("FooBar")
+          expect(ObjectInspector.inspect(user)).to eq("User (FooBar)")
+        end
+
+        it "works with subclasses" do
+          class Admin < User
+          end
+
+          admin = Admin.new("FooBar")
+          expect(ObjectInspector.inspect(admin)).to eq("User (FooBar)")
+        end
```

Comment:
There's an ordering dependency in these specs -- this example will only pass if the prior `it "works with custom class"` one runs first (since it registers the superclass).

The `register` call should be moved into a `before(:context)` hook.